lib/iglu-client/core.rb

```diff
@@ -0,0 +1,107 @@
+# Copyright (c) 2017 Snowplow Analytics Ltd. All rights reserved.
+#
+# This program is licensed to you under the Apache License Version 2.0,
+# and you may not use this file except in compliance with the Apache License Version 2.0.
+# You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the Apache License Version 2.0 is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+
+require "json-schema"
+
+module Iglu
+
+  # Regular expression to extract metadata from self-describing JSON
+  URI_REGEX = Regexp.new "^iglu:([a-zA-Z0-9\\-_.]+)\/([a-zA-Z0-9\\-_]+)\/([a-zA-Z0-9\\-_]+)\/([1-9][0-9]*(?:-(?:0|[1-9][0-9]*)){2})$"
+
+  # Regular expression to extract all parts of SchemaVer: MODEL, REVISION, ADDITION
+  SCHEMAVER_REGEX = Regexp.new "^([1-9][0-9]*)-(0|[1-9][0-9]*)-(0|[1-9][0-9]*)$"
+
+  # Class holding SchemaVer data
+  class SchemaVer
+    attr_accessor :model, :revision, :addition
+
+    # Constructor. To initalize from string - use static parse_schemaver
+    def initialize(model, revision, addition)
+      @model = model
+      @revision = revision
+      @addition = addition
+    end
+
+    # Render as string
+    def as_string
```

Comment:
Yeah, I wan't sure about it. But in the end decided that if we don't use `toString` in Scala then we probably shouldn't use standard method in other implementations.